tests/bluetooth/host/id/bt_id_read_public_addr/CMakeLists.txt

```
@@ -0,0 +1,20 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+
+FILE(GLOB SOURCES src/*.c)
```

Please don't use glob in cmake files. It makes the build system unreliable.

---

What do you think about the following for replacing `glob`?

```suggestion
set(SOURCES
  ${CMAKE_CURRENT_SOURCE_DIR}/src/main.c
  ${CMAKE_CURRENT_SOURCE_DIR}/src/test_suite_invalid_inputs.c
)

file(GLOB files src/*.c)
foreach(file ${files})
list (FIND SOURCES ${file} _index)
if(_index EQUAL -1)
  message(WARNING "File ${file} not found in sources list!")
endif()
endforeach()
```
The checking part is just for printing a warning if there are source files in the `src` directory that don't exist in the source files list.

---

I am OK with this as an experiment. The worst thing to happen would be a false warning. But we have to use `file(GLOB files CONFIGURE_DEPENDS src/*.c)` to make it a bit more reliable.

I don't like the`${CMAKE_CURRENT_SOURCE_DIR}/` in the sources list. Requiring it changes the semantics of the `SOURCES` variable (compared to the rest of Zephyr). We should be able to use relative paths there. I think we should allow relative paths and use `file(REAL_PATH ...)` in the loop to resolve them.